### ISSUE TITLE
terra-core "No Data view" (terra-clinical + terra-consumer + OCS) 

### ISSUE THREAD

**Issue by amf9t2** [cerner/terra-core#744]
# Issue Description
As an application developer, I need "No Data view" card  which will meet terra-clinical and terra-consumer requirement(theme, icon, font size etc) and also covering most of Cerner OCS No Data view card specification.

## Issue Type

- [x] New Feature
- [ ] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior

This should work similar to terra-clinical [No Data View component ](https://github.com/cerner/terra-clinical/tree/master/packages/terra-clinical-no-data-view) with theming option to satisfy terra-consumer and Cerner OCS. 

### Considerations:
- Meets specification of [No Data View component ](https://github.com/cerner/terra-clinical/tree/master/packages/terra-clinical-no-data-view) in terra-clinical 
- Meet specification of terra-consumer 
- Covers most of the Cerner one style [No Data view](http://engineering.cerner.com/terra-clinical/#/site/no-data-view) component

### Responsiveness:
- TBD

### Bidirectionality
- TBD

### Accessibility
- TBD

### React Props:
| Prop | Type | Default | Description |
|-------|-------|---------|---------------|
| heading | string |'' | The title of the no data view. |
| subtext | string |'' | The subtext description of the no data view. |
| subtextContent | element  | undefined | The element to be dispaly in the subtext content container. |
| isGlyphHidden        | bool  | false      | Whether or not the no data glyph should be displayed. |

### Example:
```
import React from 'react';
import NoDataView from 'terra-no-data-view';

<NoDataView />
<NoDataView heading={String} subtext={String} subtextContent={Button} isGlyphHidden />
<ThemeProvider themeName="terra-consumer-theme/cerner-one-style">
      <NoDataView heading={String} subtext={String} subtextContent={Button} />
 </ThemeProvider>
```
### CSS Classes:
| Selector                    | Description |
|-----------------------------|-------------|
| . no-data-view            | The root class for the CSS component. It will be applied under default theme. It is same as terra-cilinical. |
| . glyph           | The class for the glyph.It will be applied under default theme It is same as terra-cilinical. |
| . heading           | The class for the heading to change its appearance. It will be applied under default theme. It is same as terra-cilinical.|
| . subtext           |  The class for the subtext to change its appearance. It will be applied under default theme. It is same as terra-cilinical.  |
| . subtext-content           | The class for the subtext to set content position. It will be applied under default theme. It is same as terra-cilinical. |

### mixin.scss
```
@mixin terra-nodataview-default(mobile/desktop){
      $icon-size: ___; vary as per mobile/desktop
      $heading-text-size: ___; vary as per mobile/desktop/small/medium
      $heading-text-color: ___;
      $subtext-text-size: ___; 
      $subtext-text-color: ___;
      $divider-visible:_____;  
     --terra-nodataview-icon-size: var(--icon-size);
     --terra-nodataview-heading-text-size: var(--heading-text-size);
     --terra-nodataview-heading-text-color: var(--heading-text-color);
     --terra-nodataview-subtext-text-size: var(--subtext-text-size);
     --terra-nodataview-subtext-text-color: var(--subtext-text-color);
     --terra-nodataview-divider: var(--divider-visible);
}
@mixin terra-consumer-theme-nodataview(mobile/desktop){
      $icon-size: ___; vary as per mobile/desktop
      $heading-text-size: ___; vary as per mobile/desktop/small/medium
      $heading-text-color: ___;
      $subtext-text-size: ___; 
      $subtext-text-color: ___;
      $divider-visible:_____;
     --terra-nodataview-icon-size: var(--icon-size);
     --terra-nodataview-heading-text-size: var(--heading-text-size);
     --terra-nodataview-heading-text-color: var(--heading-text-color);
     --terra-nodataview-subtext-text-size: var(--subtext-text-size);
     --terra-nodataview-subtext-text-color: var(--subtext-text-color);
     --terra-nodataview-divider: var(--divider-visible);
}
```

Theme based CSS will be defined to honor variation in Icon size, heading and subtext styles. 
Theming variables will be used for different text size, text color, icon size etc.

**Comment by alex-bezek** [comment]
Looks like terra-consumer puts a horizontal rule between the heading and the subText. I'm assuming we can use a theme variable for `display: none`

Can you inquire if this component is supposed to flex between our callouts of medium/large based off breakpoints, or if they are separately exported components from this package as the Small -- component will be.

Our callouts seem to have the subtext left aligned when there are no buttons, but center aligned when there are buttons. Can you see if this is a coincidence, or needs to be configurable, or smart?

I see `No Data` and `No Matching Results`. is this some type enum we want to use, or is the `No Matching Results` really an error view component?

This component shouldn't take a theme prop. We shouldnt need to pass a different theme to the content, the whole page should be wrapped in a theme provider which will end up providing the themed variables for the content's components. 

I can't tell if OCS just has way more padding/margin or if they have a min height where consumer/clinical just wraps their content. 

Anyone have opinions on putting the `No Data` and the `Error View` in the same package and just exporting different modules as they seems to be almost identical and could share some css/base components?

**Comment by amf9t2** [comment]
@alex-bezek updated the tech design to set theming variable for divider visibility.

`No Data` and `No Matching Results` can be taken as props via heading. 

yes Removed the theme prop.

do we have the OCS component site where I can verify the padding ?

**Comment by neilpfeiffer** [comment]
@javalkar-varun @amf9t2 @alex-bezek: cross posting this same comment on both issues for coverage:
- [terra-core#744](https://github.com/cerner/terra-core/issues/744)
- [terra-core#745](https://github.com/cerner/terra-core/issues/745) duplicates [terra-consumer#16](https://github.com/cerner/terra-consumer/issues/16)
---

Let's pause on theme styling issues for a moment, I will confirm the css details needed for the theme versions next and how Legacy/Clinical/Consumer are expecting them to be presented (in a card/container vs. a built-in shape around component, etc.).

There is a good possibility we should merge `No Data` and `Error View` in one package due to matching presentation and consistency in functional API -- I want to discuss it in post-scrum or warroom Friday to confirm. But, we also may need to have two packages, one for terra-core and extended for terra-consumer OR sister packages in terra-clinical and terra-consumer (none in core) due to type differences. Here is the current list of react props I've identified after talking with both teams:

### `type` 
enum, current requirements are they will be different for consumer than clinical
- Clinical:  (1)No Data, (2)No Matching Results, (3)Not Authorized, (4)Error _/\*confirmed list\*/_
- Consumer: (1)No Data, (2)No Matching Results, (3)Not Authorized, (4)Error, (5)Sensitive Data, (6) custom _/\*team is confirming list on Friday\*/_

### `customIcon` 
element, Consumer only
- For consumer only, allowing for `type.CUSTOM` to provide a custom image

### `customTitle`
string,  _currently_ `heading`, `name`, Consumer only
- Clinical confirmed they would like the title/heading not exposed as available prop, standard translations would be provided and text gets added always based on `type`
- Consumer will be confirming if they will follow same pattern with fixed strings for their 5 defined types, allowing for react prop to be exposed but only used for `type.CUSTOM`.

### `text`
string, _currently_ `subtext`, `description`
- Both Clinical and Consumer confirmed as available prop to take a message.
- I'm wondering if actual usage will ever need to put more complex content in the message like we needed for Alert (including HTML, a reference link, etc.) and this should accept _node_ rather than _string_, but suggest leaving that for a future enhancement unless others have thoughts. Other option is to have a general `children` prop that you pass the message and any actions as a _node_

### `actions` or `children`
node, similar to current `subtextContent ` and `buttonText`+`buttonOnClick `
- needs more thinking and definition as to best way to solve, plus correct naming
- basically need one or more props to handle including multiple actions, buttons 

### `isIconHidden` 
bool, _currently_ `isGlyphHidden`, no change
- might rename to `isIconHidden` or `isImageHidden` if making new package/s _(confirming naming convention of what is a 'glyph', 'icon', 'image')_
- Both Clinical and Consumer confirmed the flexibility to hide the glyph/icon/image by manual config in addition to automatically with sizing

**Comment by tm4027** [comment]
JIRA created

**Comment by alex-bezek** [comment]
After discussing in our war room, it was decided that this component and the `Error View` https://github.com/cerner/terra-core/issues/744 component should be combined into 1 called `Status View` https://github.com/cerner/terra-core/issues/752
Closing this issue to consolidate issues into 1.
